### PR TITLE
TextField: High Contrast mode style fixes

### LINKED ISSUE #7565: TextField: Underlined/multiline/borderless: borders appear and change field height in high contrast mode.
### Environment Information 
 
 
- __Package version(s)__: >6.49.0 (all versions on fabric demo seems to be affected)
- __Browser and OS versions__: IE11/Edge in High Contrast mode

### Please provide a reproduction:

Can be seen on website when High Contrast mode is enabled.
https://fabricweb.z5.web.core.windows.net/oufr/6.119.0/#/examples/textfield
 
#### Actual behavior:

Underlined TextFields have all 4 borders, and additional bottom-border from wrapper.
Multiline and Underlined TextFields change height when changing focus/active state.
Borderless TextField gets a border when focused.
 
#### Expected behavior:
 
Unsure if this is by design, or accidental behavior. The height changes are obviously wrong, but should we keep all borders for underlined text fields, or should we strive to look as similar to non-HC mode as possible?

#### Priorities and help requested:
 
Are you willing to submit a PR to fix? Yes
 
Requested priority: Normal
 
Products/sites affected: Outlook web contact editing
 
#### Screenshots

Underlined textfields
image

Borderless with focus:
image

Jumping on focus: (Sample from multiline textfield)
peek 2019-01-09 11-52

### PR DESCRIPTION
**Pull request by Vitalius1**
#### Pull request checklist

- [X] Addresses an existing issue: Fixes #7565 
- [X] Include a change request file using `$ npm run change`

#### Description of changes

Cleaning up some border styles in the attempt to address some high contrast mode issues.
Most of the files changed are snapshots updated by removing or adding some unnecessary styles.

#### Focus areas to test

Open up the `TextField` demo page on a successful deploy link to test the HC mode.

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/TextField/TextField.styles.tsx [hunk 1]
```diff
@@ -195,7 +188,7 @@ export function getStyles(props: ITextFieldStyleProps): ITextFieldStyles {
       },
       borderless && {
         borderColor: 'transparent',
```
Comment (mikewheaton) [inline packages/office-ui-fabric-react/src/components/TextField/TextField.styles.tsx]: Is `borderColor` needed here now that `border: none` is set?

File: packages/office-ui-fabric-react/src/components/TextField/TextField.styles.tsx [hunk 2]
```diff
@@ -118,27 +105,27 @@ export function getStyles(props: ITextFieldStyleProps): ITextFieldStyles {
...
+        borderBottomColor: semanticColors.errorText,
         selectors: {
           '&:focus, &:hover': {
-            borderColor: semanticColors.errorText
+            borderBottomColor: semanticColors.errorText
           }
         }
       },
       hasErrorMessage &&
         underlined &&
```
Comment (lynamemi) [inline packages/office-ui-fabric-react/src/components/TextField/TextField.styles.tsx]: For the `hasErrorMessage && underlined` case, focus is still jumpy and doesn't have the high contrast border color.

File: packages/office-ui-fabric-react/src/components/TextField/TextField.styles.tsx (hunk 1 above)
Comment (Vitalius1) [reply packages/office-ui-fabric-react/src/components/TextField/TextField.styles.tsx]: Removed 👍

File: packages/office-ui-fabric-react/src/components/TextField/TextField.styles.tsx (hunk 2 above)
Comment (Vitalius1) [reply packages/office-ui-fabric-react/src/components/TextField/TextField.styles.tsx]: Fixed it 👍 Good catch.

**Comment by Vitalius1** [comment]
@lorejoh12 needs approval from someone on Calendar team... I made some changes to TextField styles which are used in DatePicker, so it affected some of the screenshots in that component. I did a take a look at it and visually it did not affect it.

**Comment by dzearing** [comment]
I think as long as TextField looks correct in HC, this should positively impact Calendar.

**Pull request merged**